kapitan/remoteinventory/fetch.py

```diff
@@ -0,0 +1,201 @@
+                    inv_output_path[source_uri].add(output_path)
+
+                if inv_type == "git":
+                    git_inventories[source_uri].append(inv)
+                elif inv_type in ("http", "https"):
+                    http_inventories[source_uri].append(inv)
+        except KeyError:
+            continue
+
+    git_worker = partial(fetch_git_inventories, inventory_path=inventory_path, save_dir=temp_dir)
+    http_worker = partial(fetch_http_inventories, inventory_path=inventory_path, save_dir=temp_dir)
+    [p.get() for p in pool.imap_unordered(git_worker, git_inventories.items()) if p]
+    [p.get() for p in pool.imap_unordered(http_worker, http_inventories.items()) if p]
+
+    rmtree(temp_dir)
+    logger.debug("Removed {}".format(temp_dir))
+
+
+def fetch_git_inventories(inv_mapping, inventory_path, save_dir):
+    """ Fetches remote inventories of git type as declared in target_objs
+        and recursively copies it to the output_path stored in inv_mapping
+
+        :param inv_mapping: tupple containing source and list of invetory variables
+        :param inventory_path: default or user defined inventory path
+        :param save_dir: directory to same the fetched inventory items into
+
+        :returns: None
+    """
+    source, inventories = inv_mapping
+    fetch_git_source(source, save_dir)
+    for inv in inventories:
+        repo_path = os.path.join(save_dir, os.path.basename(source))
+        repo = Repo(repo_path)
+        output_path = os.path.join(inventory_path, inv["output_path"])
+        copy_src_path = repo_path
+
+        if "ref" in inv:
+            ref = inv["ref"]
+            repo.git.checkout(ref)
+        else:
+            repo.git.checkout("master")
+
+        if "subdir" in inv:
+            sub_dir = inv["subdir"]
+            full_subdir = os.path.join(repo_path, sub_dir)
+            if os.path.isdir(full_subdir):
+                copy_src_path = full_subdir
+            else:
+                raise GitSubdirNotFoundError(
+                    "Inventory {} : subdir {} not found in repo".format(source, sub_dir)
+                )
+        # Recursively copy the fetched inventory files (src) to the output path (dst)
+        # if those items doen't exist or if it's older than src
+        copy_tree(copy_src_path, output_path, update=True)
+        logger.info("Inventory {} : saved to {}".format(source, output_path))
+
+
+def fetch_git_source(source, save_dir):
+    """Clones a git repo at source and stores it at save_dir"""
+    logger.info("Inventory {} : fetching now".format(source))
+    try:
+        Repo.clone_from(source, os.path.join(save_dir, os.path.basename(source)))
+        logger.info("Inventory {} : successfully fetched".format(source))
+    except GitCommandError as e:
+        logger.error(e)
+        raise GitFetchingError("Inventory {} : fetching unsuccessful{}".format(source, e.stderr))
+
+
+def fetch_http_inventories(inv_mapping, inventory_path, save_dir):
```

Comment:
Can't we reuse https://github.com/deepmind/kapitan/blob/7b3a4af419e9716faa11d76a6f0bc258e1b33b8b/kapitan/dependency_manager/base.py#L132 ?